### DISCUSSION TITLE
Request: Array field with set modifiers

### DISCUSSION THREAD

**Discussion by chnfyi** [pocketbase/pocketbase#6060] [Q&A]
Hi,

While _append, prepend and remove_ modifiers can be easily reproduced client side or within a custom route, it would be nice to have an array field to be able to use the built-in set modifiers that are available for other fields like multiple `select`, `relation` or `file`.

In cases we want to allow any string values, an array field would be suitable given that `select` field expects enums-like values and given that adding set modifiers support to JSON field would me more complex.

**Comment by ganigeorgiev** [comment]
It is not clear what exactly you are asking or trying to do but for now there are no plans for new "array" type field.

**Reply by chnfyi** [reply]
Ok.

To be more clear, i was just thinking about, _one day_, being able to use `something+`, `+something` and `something-` to append, prepend or removes from an array any _not-predefined_ string values.

In case of multiple values, `select` field expects predefined values, `relation` field expects record references.

For example, if we want to store a user favorites book pages or days of the year, `select` field would require to add 1, 2, 3, ... 365 as predefined expected values.

**Reply by ganigeorgiev** [reply]
It is still not clear what exactly is the use case that requires this and what problem does it actually solve. I'm not willing to add features to the roadmap if there is no clear use case or issue that the demanded feature request will solve. Based on your example you'll be better using a `relation` field and storing the data in separate collection.

**Reply by chnfyi** [reply]
For example, i tested the three options below with _myField_ as a `select` field and as a `json` field. In both cases default value is ["staff"].

```javascript

// Option 1 (sdk)
const record = await pb.collection('demo').update(id, {
    "myField+": 'editor',
})

// Option 2 (jsvm)
record.set("myField+", "editor")
$app.save(record)

// Option 3 (jsvm)
const updated = record.replaceModifiers({ "myField+": "editor" })
record.set("myField", updated.myField)
$app.save(record)

// Result for all options
// as select field: ["staff", "editor"]
// as json field: ["staff"]
```

For this case, using a `select` field is the solution. But in case we need to store an array of large possible values like, _for example_, favorite days of the year then we would have to add 365 values / choices to the list.

I understand that +/- modifiers are not currently supported for json fields which have an array as value. That's why my first idea was to have a new array type field that would not need a list of predefined values and that would support those modifiers for the convenience they provide.

_Note: Above examples are not related to some personal use cases and this request is not a current personal need, so if i failed again to be clear i won't go further and waste your time._

**Reply by javierfuentesm** [reply]
@ganigeorgiev I have another use case. I want to store values in an array; for example, I want to store multiple phone numbers related to one user, but I dont want to create a new table. I tried the select, but I have to specify the options, and I need to be open, and I dont think a JSON is needed for that. 

Is this possible?
